src/shims/env.rs

```diff
@@ -58,3 +46,81 @@ pub(crate) fn alloc_env_value<'mir, 'tcx>(
     alloc.write_bytes(&tcx, trailing_zero_ptr, &[0]).unwrap();
     ptr
 }
+
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn getenv(
+        &mut self,
+        name_op: OpTy<'tcx, Tag>,
+        dest: PlaceTy<'tcx, Tag>
```

Comment:
Can you make this return a `Scalar` instead of doing destination-passing? Same for `setenv`.

Reply:
Sure

Reply:
What about when doing `this.write_null(dest)`? Should I return an Option<Scalar> and work accordingly?

Reply:
No. Lookup `write_null`, it is defined in `helpers.rs` -- there's a scalar there as well that you can return.

Reply:
Ok, so `Scalar::from_int(0, dest.layout.size)`. Give me a couple minutes

Reply:
You'll need a size. Use `this.memory().pointer_size()`.